### PR TITLE
remove deprecations snowflake

### PR DESCRIPTION
**Pull request by Bowrna**
Related: https://github.com/apache/airflow/issues/39485
Removing deprecated SnowflakeOperator from docs and tests and replacing with SQLExecuteQueryOperator

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/system/providers/snowflake/example_snowflake.py [hunk 1]
```diff
@@ -50,23 +51,25 @@
     catchup=False,
 ) as dag:
     # [START howto_operator_snowflake]
-    snowflake_op_sql_str = SnowflakeOperator(task_id="snowflake_op_sql_str", sql=CREATE_TABLE_SQL_STRING)
+    snowflake_op_sql_str = SQLExecuteQueryOperator(
```
Comment (dirrao) [inline tests/system/providers/snowflake/example_snowflake.py:54]: Don't we need to change the `snowflake_conn_id`  key in DAG `default_args`
`default_args={"snowflake_conn_id": SNOWFLAKE_CONN_ID},`

Comment (Bowrna) [reply tests/system/providers/snowflake/example_snowflake.py:54]: yes thanks for the catch @dirrao 
I will retain both conn_id and snowflake_conn_id as the SnowflakeSqlApiOperator still supports the snowflake_conn_id

Comment (potiuk) [reply tests/system/providers/snowflake/example_snowflake.py:54]: But should not we remove the "snowflake_conn_id" completely ? Anything preventing it?

Comment (Bowrna) [reply tests/system/providers/snowflake/example_snowflake.py:54]: @potiuk `SnowflakeSqlApiOperator` is an operator to support multiple SQL statements execution sequentially. so it has inherited from` SQLExecuteQueryOperator` to support this behaviour and accepts the `snowflake_conn_id` as its param. that's why I retained both `conn_id` and `snowflake_conn_id` params in this example. Do you think we can change  `snowflake_conn_id` param in `SnowflakeSqlApiOperator` to `conn_id` itself?

https://github.com/apache/airflow/blob/d5f81a4e2de0d4236cffcf2e2d3c682b4c6ec355/airflow/providers/snowflake/operators/snowflake.py#L391-L494

Comment (potiuk) [reply tests/system/providers/snowflake/example_snowflake.py:54]: right

**Review by potiuk — APPROVED** [review]

**Pull request merged**